File: scala/src/main/scala/io/swagger/client/model/Request.scala
package io.swagger.client.model




case class Request (
  /* An array of vehicles that can be employed */
  vehicles: List[Vehicle],
  /* An array of vehicle types */
  vehicle_types: List[VehicleType],
  /* An array of services */
  services: List[Service],
  /* An array of shipments */
  shipments: List[Shipment],
  /* An array of relations */
  relations: List[Relation],
  algorithm: Algorithm)
  

File: dart/lib/model/request.dart
part of swagger.api;


@Entity()
class Request {
  /* An array of vehicles that can be employed */
  @Property(name: 'vehicles')
  List<Vehicle> vehicles = [];
  
/* An array of vehicle types */
  @Property(name: 'vehicle_types')
  List<VehicleType> vehicleTypes = [];
  
/* An array of services */
  @Property(name: 'services')
  List<Service> services = [];
  
/* An array of shipments */
  @Property(name: 'shipments')
  List<Shipment> shipments = [];
  
/* An array of relations */
  @Property(name: 'relations')
  List<Relation> relations = [];
  

  @Property(name: 'algorithm')
  Algorithm algorithm = null;
  
/* An array of objectives */
  @Property(name: 'objectives')
  List<Objective> objectives = [];
  
/* An array of cost matrices */
  @Property(name: 'cost_matrices')
  List<CostMatrix> costMatrices = [];
  
  Request();

  @override
  String toString()  {
    return 'Request[vehicles=$vehicles, vehicleTypes=$vehicleTypes, services=$services, shipments=$shipments, relations=$relations, algorithm=$algorithm, objectives=$objectives, costMatrices=$costMatrices, ]';
  }

}

